element-connector/src/main/java/org/eclipse/californium/elements/util/DatagramReader.java

```diff
@@ -180,26 +180,34 @@ public void close() {
 	 * @throws IllegalArgumentException if provided numBits exceeds available bytes
 	 */
 	public long readLong(final int numBits) {
-
+		if (numBits < 0 || numBits > Long.SIZE) {
+			throw new IllegalArgumentException("bits must be in range 0 ... 64!");
+		}
 		long bits = 0; // initialize all bits to zero
 
-		for (int i = numBits - 1; i >= 0; i--) {
-
-			// check whether new byte needs to be read
-			if (currentBitIndex < 0) {
-				readCurrentByte();
+		if (currentBitIndex < 0 && (numBits & 0x7) == 0) {
```

Comment:
`(numBits & 0x7) == 0`, I think the idea is to ensure that numbits is a multiple of 8.
For me using modulo is more intuitive.

(but probably subjective)